modules/services/servers/src/main/java/au/com/shiftyjelly/pocketcasts/servers/podcast/PodcastCacheServiceManagerImpl.kt

```diff
@@ -83,4 +85,26 @@ class PodcastCacheServiceManagerImpl @Inject constructor(
                 null
             }
         }
+
+    override suspend fun suggestedFolders(request: SuggestedFoldersRequest): List<SuggestedFolder> {
+        return service.suggestedFolders(request).toSuggestedFolders()
+    }
+
+    private fun Map<String, List<String>>.toSuggestedFolders(): List<SuggestedFolder> {
+        val folderEntities = mutableListOf<SuggestedFolder>()
+
+        this.forEach { (folderName, podcastList) ->
+            // For each podcast in the list, create a SuggestedFolder entity
+            podcastList.forEach { podcastUuid ->
+                folderEntities.add(
+                    SuggestedFolder(
+                        uuid = UUID.randomUUID().toString(),
+                        name = folderName,
+                        podcastUuid = podcastUuid,
+                    ),
+                )
+            }
+        }
+        return folderEntities
```

Comment:
This can be expressed as a regular `flatMap().map()` chaing without introducing local `val`s.